packages/wonder-blocks-data/components/track-data.md

```diff
@@ -1,3 +1,208 @@
 The `TrackData` component is a server-side only component. It should be used as
 a parent to the components whose data requests you want to fulfill during
 server-side rendering.
+
+#### Client-side behavior
+
+If used outside of server-side mode (as set using `Server.setServerSide`), this
+component will throw, as demonstrated below.
+
+```jsx
+import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
+import {Server, View} from "@khanacademy/wonder-blocks-core";
+import {TrackData} from "@khanacademy/wonder-blocks-data";
+
+class ErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    static getDerivedStateFromError(error) {
+        return {error: error.message};
+    }
+
+    render() {
+        if (jest) {
+            /**
+             * The snapshot test just sees the error getting thrown, not the
+             * awesome error boundary, so we have to hack around it to keep
+             * this live example, but not get test failures.
+             */
+            return "Sorry, no snapshot for you";
+        }
+
+        if (this.state.error) {
+            return (
+                <View>
+                    {this.state.error}
+                </View>
+            );
+        }
+        return this.props.children;
+    }
+}
+
+<ErrorBoundary>
+    <View>
+        <TrackData>
+            <Body>This only renders if we're in server-side mode and the page hot reloaded</Body>
+        </TrackData>
+    </View>
+</ErrorBoundary>
+```
+
+#### Server-side behavior
+
+When used server-side, this component tracks any data requests made through
+the `Data` component during a render cycle. This data can then be obtained
+using the `fulfillAllDataRequests` method. The data can then be used in an
+additional render cycle to render with that data.
+
+```jsx
+import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import Button from "@khanacademy/wonder-blocks-button";
+import {Server, View} from "@khanacademy/wonder-blocks-core";
+import {Data, TrackData, RequestHandler, fulfillAllDataRequests} from "@khanacademy/wonder-blocks-data";
+
+class MyPretendHandler extends RequestHandler {
+    constructor() {
+        super("MY_PRETEND_HANDLER");
```

Comment:
Is there a mechanism to verify the handlers are unique? what would happen if two different handlers happen to have the same type?

Reply:
This is mentioned in the ADR. If there are two with the same type, they will:

1. Share cache
2. When resolving on the server, the first one encountered will handle the requests.

We don't do anything to verify uniqueness. That's on the consumers of the API. In reality I don't expect any issues and this makes a much nicer API as we don't have to have some sort of "register" step for handlers.